cli_tools/common/utils/logging/service/logger.go

```diff
@@ -170,39 +169,27 @@ func getAnonymizedFailureReason(err error) string {
 	return strings.Join(anonymizedErrs, "\n")
 }
 
-func (l *Logger) getOutputInfo(w *daisy.Workflow, err error) *OutputInfo {
+func (l *Logger) getOutputInfo(loggable Loggable, err error) *OutputInfo {
 	o := OutputInfo{}
 
-	if w != nil {
-		o.TargetsSizeGb = getInt64Values(w.GetSerialConsoleOutputValue(targetSizeGb))
-		o.SourcesSizeGb = getInt64Values(w.GetSerialConsoleOutputValue(sourceSizeGb))
-		o.ImportFileFormat = w.GetSerialConsoleOutputValue(importFileFormat)
+	if loggable != nil {
+		o.TargetsSizeGb = loggable.GetValueAsInt64Slice(targetSizeGb)
+		o.SourcesSizeGb = loggable.GetValueAsInt64Slice(sourceSizeGb)
+		o.ImportFileFormat = loggable.GetValue(importFileFormat)
 	}
 
 	if err != nil {
 		o.FailureMessage = getFailureReason(err)
 		o.FailureMessageWithoutPrivacyInfo = getAnonymizedFailureReason(err)
-		if w != nil && w.Logger != nil {
-			o.SerialOutputs = w.Logger.ReadSerialPortLogs()
+		if loggable != nil {
+			o.SerialOutputs = loggable.ReadSerialPortLogs()
 		}
 	}
 
 	return &o
 }
 
-func getInt64Values(s string) []int64 {
-	strs := strings.Split(s, ",")
-	var r []int64
-	for _, str := range strs {
-		i, err := strconv.ParseInt(str, 0, 64)
-		if err == nil {
-			r = append(r, i)
-		}
-	}
-	return r
-}
-
-func (l *Logger) runWithServerLogging(function func() (*daisy.Workflow, error),
+func (l *Logger) runWithServerLogging(function func() (Loggable, error),
```

Comment:
minor: can be a good time to rename `function` to something more descriptive. Even 'funcToRun` is better than `function` which almost looks like a reserved word.